sound/soc/sof/ipc.c

```diff
@@ -247,6 +247,9 @@ int sof_ipc_tx_message(struct snd_sof_ipc *ipc, u32 header,
 	struct snd_sof_ipc_msg *msg;
 	int ret;
 
+	if (msg_bytes > PAGE_SIZE)
```

Comment:
we can probably do a better protection here. there is a define with the IPC max size somewhere, and it's way smaller than a page.

Reply:
This is a pure "mechanical" protection for memcpy(). The .msg_data is allocated with one page of memory, so, that's the check here. We could use a new macro for the IPC buffer size though. Wouldn't we need that complete page of memory with new large IPC messages?

Reply:
@lyakh the IPC is using a window allocated by the firmware, there is nothing that is tied to the Linux pages.